packages/cycle-scripts-one-fits-all/template/src/index.ts

```diff
@@ -1,37 +1,29 @@
-import xs from 'xstream';
 import { setup, run } from '@cycle/run';
-import { restartable, rerunner } from 'cycle-restart';
-import { makeDOMDriver } from '@cycle/dom';
-import { makeHTTPDriver } from '@cycle/http';
-import { timeDriver } from '@cycle/time';
+import { rerunner } from 'cycle-restart';
 import isolate from '@cycle/isolate';
 import onionify from 'cycle-onionify';
+import storageify from 'cycle-storageify';
 
-import { Component, Sources, RootSinks } from './interfaces';
+import { mkDrivers, Component } from './drivers';
 import { App } from './app';
 
-const main : Component = onionify(App);
-
-let drivers : any, driverFn : any;
-/// #if PRODUCTION
-drivers = {
-    DOM: makeDOMDriver('#app'),
-    HTTP: makeHTTPDriver(),
-    Time: timeDriver
+// TODO PR upstream - 2nd,3rd options should be optional
+declare type StorageifyOptions = {
+    key : string;
+    serialize(state : any): string;
+    deserialize(stateStr : string): any;
 };
-/// #else
-driverFn = () => ({
-    DOM: restartable(makeDOMDriver('#app'), { pauseSinksWhileReplaying: false }),
-    HTTP: restartable(makeHTTPDriver()),
-    Time: timeDriver
-});
-/// #endif
-export const driverNames : string[] = Object.keys(drivers || driverFn());
```

Comment:
Thats not an issue, without `run` a driver is just a function from sinks to sourced
If you call that function with a sink, then you may have side effects. But not before